File: frontend/src/server/classParticipant.ts
import axios from "axios";
import { API_URL } from "../env/env";
import type { ClassParticipant, ParticipantClass } from "../types/type";

export async function getClassParticipant(
  classId: number
): Promise<ClassParticipant[]> {
  try {
    const res = await axios.get(API_URL + "/enrollments/classes/" + classId);
    console.log(res.data);
    return (res.data.data as ClassParticipant[]) || [];
  } catch (error) {
    console.error(error);
    return [];
  }
}

export async function getParticipantClass(
  participantId: number
): Promise<ParticipantClass[]> {
  try {
    const res = await axios.get(
      API_URL + "/enrollments/participants/" + participantId
    );
    console.log(res.data);
    return (res.data.data as ParticipantClass[]) || [];
  } catch (error) {
    console.error(error);
    return [];
  }
}

export async function assignParticipant(
  participantId: number,
  classId: number
) {
  try {
    await axios.post(API_URL + "/enrollments/add", {
      classId: classId,
      participantId: participantId,
    });
  } catch (error) {
    console.error(error);
  }
}

export async function deleteClassParticipant(id: number) {
  try {
    await axios.delete(`${API_URL}/enrollments/${id}`);
  } catch (error) {
    console.error(error);
  }
}

File: backend/app/controller/class_participant.go
package controller

import (
	"github.com/gofiber/fiber/v2"
	"github.com/yebology/skillful-certification/app/dto/request"
	"github.com/yebology/skillful-certification/app/service"
	"github.com/yebology/skillful-certification/constant"
	"github.com/yebology/skillful-certification/output"
	"github.com/yebology/skillful-certification/utils"
)

type ClassParticipantController struct {
	service service.ClassParticipantServiceInterface
}

func NewClassParticipantController(service service.ClassParticipantServiceInterface) *ClassParticipantController {
	return &ClassParticipantController{service: service}
}

func (con *ClassParticipantController) AssignParticipant(c *fiber.Ctx) error {

	var dto request.AddClassParticipantDto

	err := utils.ParseAndValidateBody(c, &dto)
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusInternalServerError, constant.ErrParsingBody, nil)
	}

	err = con.service.AssignParticipantService(dto)
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusInternalServerError, err.Error(), nil)
	}

	return output.GetOutput(c, constant.StatusSuccess, fiber.StatusOK, constant.SuccessAssignParticipantToClass, nil)

}

func (con *ClassParticipantController) GetParticipantClass(c *fiber.Ctx) error {

	participantId, err := utils.ConvertToNum(c, "participant_id")
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusInternalServerError, constant.ErrConvertNum, nil)
	}

	classes, err := con.service.FetchParticipantClassService(participantId)
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusNotFound, err.Error(), nil)
	}

	return output.GetOutput(c, constant.StatusSuccess, fiber.StatusOK, constant.SuccessFetchParticipantClass, classes)

}

func (con *ClassParticipantController) GetClassParticipant(c *fiber.Ctx) error {

	classId, err := utils.ConvertToNum(c, "class_id")
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusInternalServerError, constant.ErrConvertNum, nil)
	}

	participants, err := con.service.FetchClassParticipantService(classId)
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusNotFound, err.Error(), nil)
	}

	return output.GetOutput(c, constant.StatusSuccess, fiber.StatusOK, constant.SuccessFetchClassParticipants, participants)

}

func (con *ClassParticipantController) DeleteClassParticipant(c *fiber.Ctx) error {

	classParticipantId, err := utils.ConvertToNum(c, "id")
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusInternalServerError, constant.ErrConvertNum, nil)
	}

	err = con.service.DeleteClassParticipantService(classParticipantId)
	if err != nil {
		return output.GetOutput(c, constant.StatusError, fiber.StatusNotFound, err.Error(), nil)
	}

	return output.GetOutput(c, constant.StatusSuccess, fiber.StatusOK, constant.SuccessDeleteEnrollment, nil)

}
